### PR TITLE
Resolve singletons on non-authoritative servers

### PR DESCRIPTION
**Pull request by improbable-valentyn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Singletons were not resolved on non-authoritative servers because the call to `ResolveEntityActor` was removed from the bit in `ReceiveActor` where the actor already exists.
#### Primary reviewers
@Vatyx @m-samiec

### CODE REVIEW

File: SpatialGDK/Source/Public/Interop/SpatialStaticComponentView.h [hunk 1]
```diff
@@ -23,13 +23,15 @@ class SPATIALGDK_API USpatialStaticComponentView : public UObject
...
-	template <class T> T* GetComponentData(Worker_EntityId EntityId);
+	template <typename T>
+	T* GetComponentData(Worker_EntityId EntityId);
 
 	void OnAddComponent(const Worker_AddComponentOp& Op);
 	void OnRemoveEntity(const Worker_RemoveEntityOp& Op);
 	void OnComponentUpdate(const Worker_ComponentUpdateOp& Op);
 	void OnAuthorityChange(const Worker_AuthorityChangeOp& Op);
 
+private:
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/Public/Interop/SpatialStaticComponentView.h]: This is a drive-by style fix.

File: SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp [hunk 2]
```diff
@@ -75,15 +75,8 @@ void UGlobalStateManager::LinkExistingSingletonActors()
...
-		improbable::UnrealMetadata* UnrealMetadata = StaticComponentView->GetComponentData<improbable::UnrealMetadata>(SingletonEntityId);
-		if (UnrealMetadata == nullptr)
-		{
-			// Don't have entity checked out
-			continue;
-		}
-
 		// Since the entity already exists, we have to handle setting up the PackageMap properly for this Actor
-		NetDriver->PackageMap->ResolveEntityActor(SingletonActor, SingletonEntityId, UnrealMetadata->SubobjectNameToOffset);
+		NetDriver->PackageMap->ResolveEntityActor(SingletonActor, SingletonEntityId, improbable::CreateOffsetMapFromActor(SingletonActor));
```
Comment (Vatyx) [inline SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp]:
```suggestion
		asdfjashdfjklhasdkljfhaslidkfhqjkwebrqnm,webrqwerpuqhwuiefhgasdzhjklcbvzxckjvzxcv
```

Comment (improbable-valentyn) [reply SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp]: 👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏
👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏👏

Comment (Vatyx) [reply SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp]: Valentyn might remove these comments if you have too much fun. Watch out
